File: typescript-project-euler-100/src/p002.ts
import { Solution } from "./Solution";

export class P002 extends Solution{

    solve() {
        this.problemNumber = 2;
        const max = 4000000;
        let previous = 1;
        let current = 2;
        let sum = current;

        while (current < max) {
            let next = previous + current;
            if ((next < max) && (next % 2 == 0))
                sum += next;
            previous = current;
            current = next;
        }

        return sum;
    }
}

File: csharp-project-euler-100/src/P002.cs


namespace net.talaatharb.projecteuler100
{

    /**
    * @author mharb
    *
    */
    public class P002 : Solution
    {

        /**
         * Problem solution entry point
         * 
         * @param args
         */
        public static void main(String[] args)
        {
            new P002().run();
        }

        public override int solve()
        {
            problemNumber = 2;
            long max = 4000000;
            int previous = 1;
            int current = 2;
            int sum = current;
            while (current < max)
            {
                int next = previous + current;
                if ((next < max) && (next % 2 == 0))
                {
                    sum += next;
                }
                previous = current;
                current = next;
            }

            return sum;
        }
    }
}

